src/app/interaction-model/InvokeResponder.h

```diff
@@ -0,0 +1,316 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/MessageDef/StatusIB.h>
+#include <app/data-model/WrappedStructEncoder.h>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace InteractionModel {
+
+/// Handles encoding of an invoke response for a specific invoke request.
+///
+/// This class handles a single response (i.e. a CommandDataIB within the
+/// matter protocol) and is responsible for constructing its corresponding
+/// response (i.e. a InvokeResponseIB within the matter protocol)
+///
+/// Invoke responses MUST contain exactly ONE of:
+///   - response data (accessed via `ResponseEncoder`)
+///   - A status, which may be success or failure, both of which may
+///     contain a cluster-specific error code.
+///
+/// To encode a response, `Complete` MUST be called.
+///
+/// `Complete` requirements
+///   - Complete with InteractionModel::Status::Success will respond with data
+///     some response data was written.
```

Comment:
I think we will iterate a bit here: I wanted to have a separate class with a more ergonomic design, however it seems that it would require sigificant changes to integrate into the existing APIs.

As a result, I will move invoke to use some `CommandHandler` usage instead, so this class becomes obsolete. All implementation logic in it then will go away, so will not try hard to update it.

Reply:
I plan to change the invoke calls to be passed through a `CommandHandler` so I will not try to update InvokeResponder at all for now, will do that as a separate path towards CommandHandler.